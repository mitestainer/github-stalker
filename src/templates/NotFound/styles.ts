import styled, { css } from 'styled-components'
import { Page as PageStyles } from 'templates/Home/styles'
import { Container } from 'components/Container'

export const Page = styled(PageStyles)`
  max-width: 75rem;
  width: 100%;

  ${Container} {
    width: 100%;
  }
`

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    row-gap: ${theme.gutter};
    background-color: ${theme.colors.white};
    padding: ${theme.gutter};
    text-align: center;
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.darkPurple};
  `}
`
