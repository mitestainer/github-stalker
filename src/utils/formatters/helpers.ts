export const formatDate = (dateString: string) => {
  return new Intl.DateTimeFormat('en-US', {
    day: 'numeric',
    month: 'short',
    year: 'numeric'
  }).format(new Date(dateString))
}
