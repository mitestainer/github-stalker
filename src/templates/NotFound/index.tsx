import { Container } from 'components/Container'
import SearchBar from 'components/SearchBar'
import Image from 'next/image'
import * as S from './styles'

const NotFound = () => {
  return (
    <S.Page>
      <Container>
        <S.Wrapper>
          <S.Title>It happens in the best of families...</S.Title>
          <Image
            src="/img/404.svg"
            alt="404 - Not found"
            height={200}
            width={200}
          />
          <p>
            {`Seems this organization doesn't exist! Try searching for another one:`}
          </p>
          <SearchBar />
        </S.Wrapper>
      </Container>
    </S.Page>
  )
}

export default NotFound
