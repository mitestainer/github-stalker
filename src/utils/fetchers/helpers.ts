import { formatDate } from 'utils/formatters/helpers'

type RepoResponseProps = {
  id: number
  full_name: string
  description: string
  html_url: string
  watchers: number
  language: string
  updated_at: string
}

export const fetchRepos = async (handle: string, page: number) => {
  const response = await fetch(
    `https://api.github.com/orgs/${handle}/repos?sort=updated&direction=desc&page=${page}&per_page=12`
  )
  const data = await response.json()

  return data.map((repo: RepoResponseProps) => ({
    id: repo.id,
    name: repo.full_name,
    description: repo.description,
    url: repo.html_url,
    stars: repo.watchers,
    language: repo.language,
    updatedAt: formatDate(repo.updated_at)
  }))
}
