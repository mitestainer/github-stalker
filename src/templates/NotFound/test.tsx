import { render, screen } from 'utils/tests/helpers'

import NotFound from '.'

jest.mock('components/SearchBar', () => {
  return {
    __esModule: true,
    default: function () {
      return <div data-testid="Mock SearchBar"></div>
    }
  }
})

describe('<NotFound />', () => {
  it('should render the 404 picture and the search bar', () => {
    const { container } = render(<NotFound />)

    expect(
      screen.getByRole('heading', {
        name: /it happens in the best of families/i
      })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /404 - not found/i })
    ).toBeInTheDocument()
    expect(
      screen.getByText(/this organization doesn't exist/i)
    ).toBeInTheDocument()
    expect(screen.getByTestId('Mock SearchBar')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
