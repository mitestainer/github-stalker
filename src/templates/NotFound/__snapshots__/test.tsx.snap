// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render the 404 picture and the search bar 1`] = `
.c3 {
  max-width: 120rem;
  margin: 0 auto;
  background-color: #eff1fa;
  padding: 2rem;
  border-radius: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  margin: 0 auto;
}

.c1 {
  max-width: 75rem;
  width: 100%;
}

.c1 .c2 {
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 2rem;
  background-color: #fff;
  padding: 2rem;
  text-align: center;
}

.c5 {
  color: #5f49ca;
}

@media screen and (max-width:768px) {
  .c3 {
    max-width: 100%;
  }
}

@media screen and (max-width:768px) {
  .c0 .c2 {
    margin: auto 2rem;
  }
}

<main
  class="c0 c1"
>
  <div
    class="c2 c3"
  >
    <div
      class="c4"
    >
      <h1
        class="c5"
      >
        It happens in the best of families...
      </h1>
      <img
        alt="404 - Not found"
        height="200"
        src="/img/404.svg"
        width="200"
      />
      <p>
        Seems this organization doesn't exist! Try searching for another one:
      </p>
      <div
        data-testid="Mock SearchBar"
      />
    </div>
  </div>
</main>
`
